client/src/embedded_support.ts

```diff
@@ -77,18 +88,44 @@ function isPropertyAssignmentToStringOrStringInArray(
     const isCursorInToken = scanner.getStartPos() <= offset &&
         scanner.getStartPos() + scanner.getTokenText().length >= offset;
     if (propertyAssignmentContext && isCursorInToken && isStringToken) {
-      return true;
+      const start = scanner.getTokenPos();
+      const end = start + scanner.getTokenText().length;
+      return {start, end};
     }
 
     lastTokenText = scanner.getTokenText();
     lastToken = token;
     token = scanner.scan();
   }
 
-  return false;
+  return undefined;
 }
 
 function isPropertyAssignmentTerminator(token: ts.SyntaxKind) {
   return token === ts.SyntaxKind.EndOfFileToken || token === ts.SyntaxKind.CommaToken ||
       token === ts.SyntaxKind.SemicolonToken || token === ts.SyntaxKind.CloseBraceToken;
+}
+
+/** Takes a TS file and strips out all non-inline template content */
```

Comment:
Could you please explain the reasoning for doing so? I'm a little lost here..

Reply:
Yea, so we need to create a virtual document with only HTML content but preserve the position of the content for when we forward the request to the other providers. So this function finds the inline template and includes only that while replacing everything else with whitespace to preserve those positions.